pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -0,0 +1,256 @@
+    Therefore we can define a cost by:
+
+    .. math::
+        C_{HST} = 1 - \frac{1}{d^2} \left|Tr(V^{\dagger}U)\right|^2,
+
+    where the quantity :math:`\frac{1}{d^2} \left|Tr(V^{\dagger}U)\right|^2` is the Hilbert Schmidt Test. It is
+    equivalent to taking the outcome probability of the state :math:`|0 ... 0\rangle` for the following circuit:
+
+    .. figure:: ../../_static/templates/subroutines/hst.png
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    It defines our decomposition for the Hilbert Schmidt Test template.
+
+    Args:
+        params (array): Parameters for the quantum function `V`.
+        v_function (Callable): Quantum function that represents the approximate compiled unitary `V`.
+        v_wires (int or Iterable[Number, str]]): the wire(s) the approximate compiled unitary act on.
+        u_tape (.QuantumTape): `U`, the unitary to be compiled as a ``qml.tape.QuantumTape``.
+
+    Raises:
+        QuantumFunctionError: The argument u_tape must be a QuantumTape.
+        QuantumFunctionError: ``v_function`` is not a valid Quantum function.
+        QuantumFunctionError: `U` and `V` do not have the same number of wires.
+        QuantumFunctionError: The wires ``v_wires`` are a subset of `V` wires.
+        QuantumFunctionError: u_tape and v_tape must act on distinct wires.
+
+    .. UsageDetails::
+
+        Consider that we want to evaluate the Hilbert Schmidt Test cost between the unitary `U` and an approximate
+        unitary `V`. We need to define some functions where it is possible to use the ``.HilbertSchmidt`` template.
+        Here the considered unitary is ``Hadamard`` and we try to compute the cost for the approximate unitary ``RZ``.
+        For an angle which is equal to 0 (identity), we have the maximal cost which is 1.
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import HilbertSchmidt
+
+            with qml.tape.QuantumTape(do_queue=False) as U:
+                qml.Hadamard(wires=0)
+
+            def v_circuit(params, v_wires):
+                qml.RZ(params[0], wires=v_wires[0])
+
+            dev = qml.device("default.qubit", wires=4)
+
+            @qml.qnode(dev)
+            def hilbert_test(v_params, v_function, v_wires, u_tape):
+                qml.HilbertSchmidt(v_params, v_function=v_function, v_wires=v_wires, u_tape=u_tape)
+                return qml.probs(u_tape.wires + v_wires)
+
+            def cost_hst(parameters, v_function, v_wires, u_tape):
+                return (1 - hilbert_test(v_params=parameters, v_function=v_function, v_wires=v_wires, u_tape=u_tape)[0])
+
+        Now that you have defined your cost function you can call it for specific parameters.
+
+        >>> cost_hst([0], v_function = v_circuit, v_wires = [1], u_tape = U)
+        1
+
+
+    **Reference**
```

Comment:
How about moving this above the `UsageDetails` section? Right now, it renders a bit odd:

![image](https://user-images.githubusercontent.com/24476053/163210732-85d76d08-52e2-4c5a-9849-19be981b40b8.png)

Reply:
@antalszava Or move it before usage details?